### PR TITLE
One Time Checkout - Recaptcha required to submit form

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
## What are you doing in this PR?

Existing Checkout works great with Stripe, can see payments appearing in [Stripe Test](https://dashboard.stripe.com/test/payments).

Adds RECAPTCHA as required field before you can submit form for payment.

[Trello Card](https://trello.com/c/RU9Pes3Q/1077-generic-checkout-one-time-implement-express-payment-options-for-one-time-note-use-one-time-stripe-accounts)

image

### CODE REVIEW

**Review by jamesgorrie — COMMENTED** [review]
What's the UX of this when it's not checked? Do we let the person know what's missing?

**Comment by paul-daniel-dempsey** [comment]
> What's the UX of this when it's not checked? Do we let the person know what's missing?

It behaves the same as the current generic checkout. There is no validation warnings on the credit card fields in either. This stops the checkout locking up if you dont press it (easily done). Can create a seperate card to add validation to both if you wish?

File: support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx [hunk 1]
```diff
@@ -470,8 +470,7 @@ function OneTimeCheckoutComponent({
 																	setRecaptchaToken(token);
 																}}
 																onRecaptchaExpired={() => {
-																	// no-op
-																	// ToDo - Should we not expire this?
+																	setRecaptchaToken(undefined);
```
Comment (andrewHEguardian) [inline support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx:473]: Should we do the same thing on the generic checkout?

**Review by andrewHEguardian — APPROVED** [review]
Nice 👍

File: support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx (hunk 1 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx:473]: Have never used this before, maybe do a separate PR so can revert easily?

**Pull request merged**